### PR TITLE
[RNMobile] Consolidate and clarify React Native documentation

### PR DESCRIPTION
**Pull request by dcalhoun**
## Description

Rewrite the `react-native-editor` README to more accurately describe the package's purpose and consolidate duplicative documentation into a single location to avoid stale documentation. These changes include a link to the previous content, which is also found in [the top-level documentation directory](https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/getting-started-native-mobile.md).

The catalyst for these changes was that newcomers to the project have occasionally mistakenly executed the
provided commands within the nested `react-native-editor` directory.

## How has this been tested?

Visit relevant documentation file pages on GitHub to verify accurate content and working links.

## Screenshots 

n/a

## Types of changes

Documentation

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by dcalhoun** [comment]
@oguzkocer would you please review these changes to verify they help provide clarity on setting up the Gutenberg React Native development environment? I was informed you may have encountered issues running the initial project setup commands and hoped these change may help future newcomers. I'd appreciate your input. Thanks!

@jd-alexander would you mind reviewing these changes to ensure they do not introduce any confusion or inaccuracies? Thanks!

**Review by oguzkocer — APPROVED** [review]
Thanks for these changes @dcalhoun! The additional notes about running the commands from the top-level directory helps.

One thing I wanted to ask: why do we prefer `nvm install --latest-npm` command over `nvm use`? I understand that the `.nvmrc` doesn't have a specific version right now, which is a little confusing to me, but in case the project starts using a set version `nvm use` would be better, right? If so, do you think it makes sense to document it as such since that command would work either way?

**Comment by dcalhoun** [comment]
> One thing I wanted to ask: why do we prefer `nvm install --latest-npm` command over `nvm use`?

@oguzkocer from my understanding of how [the `.nvmrc` file](https://github.com/nvm-sh/nvm/blob/master/README.md#nvmrc) works, one can use both `nvm use` or `nvm install`. Both will read from the `.nvmrc` file. However, I do not believe the former will _install_ a Node.js version if it _is not already_ installed on the system, `nvm use` merely sets the "active" Node.js version being used. I believe that may be why the "getting started" documentation directs one to run `nvm install`. If you already have the relevant version installed with `nvm`, you likely could run `nvm use` instead.

> I understand that the `.nvmrc` doesn't have a specific version right now, which is a little confusing to me, but in case the project starts using a set version `nvm use` would be better, right? If so, do you think it makes sense to document it as such since that command would work either way?

The projects `.nvmrc` file leverages [an alias for the latest LTS release](https://github.com/nvm-sh/nvm/blob/master/README.md#long-term-support). That is why there is not a "specific version" found in the file. Using the alias allows the developer to leverage the latest version of the LTS, rather than the team having to manually bump the version. I do not believe declaring a "specific version" would have an impact on the differences between `nvm use` and `nvm install`.

Hope this helps!

File: packages/react-native-editor/README.md [hunk 1]
```diff
@@ -1,141 +1,10 @@
-# Mobile Gutenberg
+# React Native Editor
 
-This is the mobile version of [Gutenberg](https://github.com/WordPress/gutenberg), targeting Android and iOS. It's a React Native library bootstrapped by CRNA and now ejected.
+This package provides a demo application to simplify environment setup required for development of Gutenberg for native Android and iOS. The demo application allows running the mobile versions of Gutenberg blocks, while avoiding the additional setup steps required by the [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android) and [Wordpress iOS](https://github.com/wordpress-mobile/WordPress-iOS) apps.
```
Comment (jd-alexander) [inline packages/react-native-editor/README.md:3]: A few suggestions to improve how the sentences read :) 

```suggestion
This package provides a demo application to simplify the environment setup required for the development of Gutenberg for native Android and iOS. The demo application allows running the mobile versions of Gutenberg blocks while avoiding the additional setup steps required by the [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android) and [Wordpress iOS](https://github.com/wordpress-mobile/WordPress-iOS) apps.
```

**Review by jd-alexander — APPROVED** [review]
Thanks for these improvements @dcalhoun LGTM 🚢

**Pull request merged**